tracing/tracer.go

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package tracing
+
+import (
+	"context"
+
+	"github.com/urfave/cli/v2"
+	"go.opentelemetry.io/otel"
+	otlp "go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+)
+
+// initTracer returns the tracer provider supplied to the tracing config.
+func initTracer(c *cli.Context) (*sdktrace.TracerProvider, error) {
+	client := otlptracehttp.NewClient()
+	ctx := context.Background()
+
+	// TODO: inject actual context
+	exporter, err := otlp.New(ctx, client)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := resource.New(ctx, resource.WithAttributes(
+		semconv.ServiceName(c.String("tracing.service.name")),
+	))
+	if err != nil {
+		return nil, err
+	}
+
+	tp := sdktrace.NewTracerProvider(
+		sdktrace.WithSampler(sdktrace.AlwaysSample()),
```

Comment:
should we use RatioBased sampling? AlwaysSample seems like it might cause excessive overhead.

or... we could somehow expose this as something changeable through platform settings to swap it without rebooting. thinking, in the event tracing causes too much overhead and we want to disable it until we need it